front/components/assistant/conversation/RetrievalAction.tsx

```diff
@@ -0,0 +1,155 @@
+  github: "/static/github_black_32x32.png",
+};
+
+export default function RetrievalAction({
+  retrievalAction,
+  status,
+}: {
+  retrievalAction: RetrievalActionType;
+  status: "loading" | "retrieving" | "success" | "error";
+}) {
+  const { query, relativeTimeFrame } = retrievalAction.params;
+  function shortText(text: string, maxLength = 20) {
+    return text.length > maxLength
+      ? text.substring(0, maxLength) + "..."
+      : text;
+  }
+  if (status === "loading") {
+    return (
+      <div>
+        <div className="mb-2 text-xs font-bold text-element-600">
+          I'm thinking...
+        </div>
+        <Spinner size="sm" />
+      </div>
+    );
+  }
+  if (status === "error") {
+    return (
+      <div>
+        <div className="mb-2 text-xs font-bold text-element-600">
+          Something went wrong when searching for documents...
+        </div>
+      </div>
+    );
+  }
+  return (
+    <div>
+      <div className="mb-2 text-xs font-bold text-element-600">
+        Searching for:
+      </div>
+      <Chip.List>
+        <Tooltip label="Docs created or updated during that time are included in the search">
+          <Chip
+            color="amber"
+            label={
+              relativeTimeFrame
+                ? "During the last " +
+                  (relativeTimeFrame.duration > 1
+                    ? `${relativeTimeFrame.duration} ${relativeTimeFrame.unit}s`
+                    : `${relativeTimeFrame.unit}`)
+                : "All time"
+            }
+          />
+        </Tooltip>
+        <Tooltip label={`Query used for semantic search: ${query}`}>
+          <Chip color="slate" label={query ? shortText(query) : "No query"} />
+        </Tooltip>
+      </Chip.List>
+      {status === "retrieving" ? (
+        <div>
+          <div className="my-2 text-xs font-bold text-element-600">
+            Retrieving...
+          </div>
+          <Spinner size="sm" />
+        </div>
+      ) : (
+        <div className="my-2 text-xs font-bold text-element-600">
+          Retrieved:
+          <Chip color="violet" className="ml-2">
+            {retrievalAction.documents.length > 0
+              ? RetrievedDocumentsInfo(retrievalAction.documents)
+              : "No documents found"}
+          </Chip>
+        </div>
+      )}
+    </div>
+  );
+}
+
+function RetrievedDocumentsInfo(documents: RetrievalDocumentType[]) {
+  const summary = documentsSummary(documents);
+  return (
+    <>
+      <span>{documents.length} results</span>
+      {Object.keys(summary).map((k) => {
+        return (
+          <div key={k} className="ml-3 flex flex-initial flex-row items-center">
+            <div className={classNames("mr-1 flex h-4 w-4")}>
+              {summary[k].provider !== "none" ? (
+                <img src={PROVIDER_LOGO_PATH[summary[k].provider]}></img>
+              ) : (
+                <DocumentDuplicateIcon className="h-4 w-4 text-slate-500" />
```

Comment:
I think it's probably the wrong icon. Did you check with @Duncid ?

Reply:
I will